src/agent0/chainsync/db/hyperdrive/interface.py

```diff
@@ -47,50 +111,150 @@ def add_transfer_events(transfer_events: list[TradeEvent], session: Session) ->
-        The wallet address to filter the results on.
+    wallet_addr: str | None
+        The wallet address to filter the results on. Can be None to return latest block number
+        regardless of wallet.
+    hyperdrive_address: str | None
+        The hyperdrive address to filter the results on. Can be None to return latest block number
+        regardless of pool.
+
+    Returns
+    -------
+    int
+        The latest block number in the hyperdrive_events table.
+    """
+
+    query = session.query(func.max(TradeEvent.block_number))
+    if wallet_addr is not None:
+        query = query.filter(TradeEvent.wallet_address == wallet_addr)
+    if hyperdrive_address is not None:
+        query = query.filter(TradeEvent.hyperdrive_address == hyperdrive_address)
+    query = query.scalar()
+
+    if query is None:
+        return 0
+    return int(query)
+
+
+def get_latest_block_number_from_positions_snapshot_table(
+    session: Session, wallet_addr: str | None, hyperdrive_address: str | None
+) -> int:
+    """Get the latest block number based on the positions snapshot table in the db.
+
+    Arguments
+    ---------
+    session: Session
+        The initialized session object.
+    wallet_addr: str | None
+        The wallet address to filter the results on. Can be None to return latest block number
+        regardless of wallet.
+    hyperdrive_address: str | None
+        The hyperdrive address to filter the results on. Can be None to return latest block number
+        regardless of pool.
 
     Returns
     -------
     int
         The latest block number in the hyperdrive_events table.
     """
 
-    query = session.query(func.max(TradeEvent.block_number)).filter(TradeEvent.wallet_address == wallet_addr).scalar()
+    query = session.query(func.max(PositionSnapshot.block_number))
+    if wallet_addr is not None:
+        query = query.filter(PositionSnapshot.wallet_address == wallet_addr)
+    if hyperdrive_address is not None:
+        query = query.filter(PositionSnapshot.hyperdrive_address == hyperdrive_address)
+    query = query.scalar()
+
     if query is None:
         return 0
     return int(query)
 
 
-def get_trade_events(session: Session, wallet_addr: str | None = None) -> pd.DataFrame:
+def get_trade_events(
+    session: Session,
+    wallet_address: str | list[str] | None = None,
+    hyperdrive_address: str | None = None,
+    all_token_deltas: bool = True,
+    sort_ascending: bool = True,
+    query_limit: int | None = None,
+    coerce_float=False,
+) -> pd.DataFrame:
     """Get all trade events and returns a pandas dataframe.
 
     Arguments
     ---------
     session: Session
         The initialized db session object.
-    wallet_addr: str | None, optional
-        The wallet address to filter the results on. Return all if None.
+    wallet_address: str | list[str] | None, optional
+        The wallet address(es) to filter the results on. Return all if None.
+    hyperdrive_address: str | None, optional
```

Comment:
update these comments to have `, optional` where appropriate and specify the default.